src/game_save.cpp

```diff
@@ -807,7 +807,7 @@ bool loadGame(bool &generate) {
             count = rdByte();
             char_tmp = rdByte();
             for (int i = count; i > 0; i--) {
-                if (tile >= &dg.floor[MAX_HEIGHT][0]) {
+                if (tile >= &dg.floor[MAX_HEIGHT - 1][0]) {
```

Comment:
Thanks for the PR Dan. On the surface, the change makes sense, however, I don't believe this cave reading code will work any more.

The `goto error` will be triggered when the `tile` reference is `dg.floor[MAX_HEIGHT - 1][1]`, meaning the last row of data will not be read.

When the `tile >=` conditional is checked, the previous `tile++` means the reference will be greater than `dg.floor[MAX_HEIGHT - 1][MAX_WIDTH - 1]`, so even though `dg.floor[MAX_HEIGHT][0]` is an out of bounds access, it never actually tries to write to it. The function works, even though it's potentially dangerous!

I think this section of the code therefore needs a complete rewrite.